cdc/processor/pipeline/table_actor.go

```diff
@@ -0,0 +1,436 @@
+
+	t.started = true
+	log.Info("table actor is started",
+		zap.String("tableName", t.tableName),
+		zap.Int64("tableID", t.tableID))
+	return nil
+}
+
+// stop the actor, it's idempotent
+func stop(ctx context.Context, t *tableActor) func(err error) {
+	return func(err error) {
+		t.lck.Lock()
+		defer t.lck.Unlock()
+
+		if atomic.LoadUint32(&t.stopped) == stopped {
+			return
+		}
+		atomic.StoreUint32(&t.stopped, stopped)
+		t.cancel()
+		if t.sortNode != nil {
+			t.sortNode.ReleaseResource(ctx, t.changefeedID, t.globalVars.CaptureInfo.AdvertiseAddr)
+		}
+		if t.sinkNode != nil {
+			if err := t.sinkNode.ReleaseResource(ctx); err != nil {
+				log.Warn("close sink failed",
+					zap.String("changefeed", t.changefeedID),
+					zap.String("tableName", t.tableName),
+					zap.Error(err), zap.Error(err))
+			}
+		}
+		log.Info("table actor will be stopped",
+			zap.String("changefeed", t.changefeedID),
+			zap.String("tableName", t.tableName),
+			zap.Int64("tableID", t.tableID),
+			zap.Error(err))
+	}
+}
+
+// any error will be reported to processor
+func (t *tableActor) checkError(err error) {
+	t.stopFunc(err)
+	t.reportErr(err)
+}
+
+// ============ Implement TablePipline, must be threadsafe ============
+
+// ResolvedTs returns the resolved ts in this table pipeline
+func (t *tableActor) ResolvedTs() model.Ts {
+	// TODO: after TiCDC introduces p2p based resolved ts mechanism, TiCDC nodes
+	// will be able to cooperate replication status directly. Then we will add
+	// another replication barrier for consistent replication instead of reusing
+	// the global resolved-ts.
+	if redo.IsConsistentEnabled(t.replConfig.Consistent.Level) {
+		return t.sinkNode.ResolvedTs()
+	}
+	return t.sortNode.ResolvedTs()
+}
+
+// CheckpointTs returns the checkpoint ts in this table pipeline
+func (t *tableActor) CheckpointTs() model.Ts {
+	return t.sinkNode.CheckpointTs()
+}
+
+// UpdateBarrierTs updates the barrier ts in this table pipeline
+func (t *tableActor) UpdateBarrierTs(ts model.Ts) {
+	msg := message.BarrierMessage(ts)
+	err := t.tableActorRouter.Send(t.actorID, msg)
+	if err != nil {
+		log.Warn("send fails",
+			zap.Reflect("msg", msg),
+			zap.String("tableName", t.tableName),
+			zap.Int64("tableID", t.tableID),
+			zap.Error(err))
+	}
+}
+
+// AsyncStop tells the pipeline to stop, and returns true if the pipeline is already stopped.
+func (t *tableActor) AsyncStop(targetTs model.Ts) bool {
+	// TypeStop stop the sinkNode only ,the processor stop the sink to release some resource
+	// and then stop the whole table pipeline by call Cancel
+	msg := message.StopMessage()
+	err := t.tableActorRouter.Send(t.actorID, msg)
+	log.Info("send async stop signal to table",
+		zap.String("tableName", t.tableName),
+		zap.Int64("tableID", t.tableID),
+		zap.Uint64("targetTs", targetTs))
+	if err != nil {
+		if cerror.ErrMailboxFull.Equal(err) {
+			return false
+		}
+		if cerror.ErrSendToClosedPipeline.Equal(err) {
```

Comment:
This err should be `cerrors.ErrActorNotFound`.

Reply:
fixed